src/app/components/core/asset-readings/readings-graph/readings-graph.component.ts

```diff
@@ -1,61 +1,156 @@
+    displaylogo: false,
+    displayModeBar: true,
+    modeBarButtonsToRemove: ['resetScale2d', 'hoverClosestCartesian', 'select2d',
+      'hoverCompareCartesian', 'lasso2d', 'zoom2d', 'autoScale2d', 'pan2d',
+      'zoomIn2d', 'zoomOut2d', 'toImage', 'toggleSpikelines'],
+    modeBarButtonsToAdd: [[
+      {
+        name: 'Zoom in',
+        icon: {
+          'width': 875,
+          'height': 1000,
+          'path': 'm1 787l0-875 875 0 0 875-875 0z m687-500l-187 0 0-187-125 0 0 187-188 0 0 125 188 0 0 187 125 0 0-187 187 0 0-125z',
+          'transform': 'matrix(1 0 0 -1 0 850)'
+        },
+        click: () => {
+          if (this.timeWindowIndex <= 0) {
+            console.log('minimum zoom level reached');
+            return;
+          }
+
+          if (!this.zoomOut && (this.numReadings.length <= 0 || (this.numReadings.length > 0 && this.numReadings[0].x.length <= 1))) {
+            console.log('No readings to zoom in');
+            return;
+          }
+          this.zoomOut = false;
+          this.timeWindowIndex--;
+          const timeWindow = Utils.getTimeWindow(this.timeWindowIndex);
+          if (this.panning) {
+            this.panModeZoom();
+            return;
+          }
+          this.zoomGraph(timeWindow.value);
+        }
+      },
+      {
+        name: 'Zoom out',
+        icon: {
+          'width': 875,
+          'height': 1000,
+          'path': 'm0 788l0-876 875 0 0 876-875 0z m688-500l-500 0 0 125 500 0 0-125z',
+          'transform': 'matrix(1 0 0 -1 0 850)'
+        },
+        click: () => {
+          this.zoomOut = true;
+          if (this.timeWindowIndex >= Utils.getTimeWindow(this.timeWindowIndex).size - 1) {
+            console.log('maximum zoom level reached');
+            return;
+          }
+          this.timeWindowIndex++;
+          const timeWindow = Utils.getTimeWindow(this.timeWindowIndex);
+          if (this.panning) {
+            this.panModeZoom(true);
+            return;
+          }
+          this.zoomGraph(timeWindow.value);
+        }
+      },
+      {
+        name: 'Reset',
+        icon: {
+          'width': 1000,
+          'height': 1000,
+          // tslint:disable-next-line: max-line-length
+          'path': 'm250 850l-187 0-63 0 0-62 0-188 63 0 0 188 187 0 0 62z m688 0l-188 0 0-62 188 0 0-188 62 0 0 188 0 62-62 0z m-875-938l0 188-63 0 0-188 0-62 63 0 187 0 0 62-187 0z m875 188l0-188-188 0 0-62 188 0 62 0 0 62 0 188-62 0z m-125 188l-1 0-93-94-156 156 156 156 92-93 2 0 0 250-250 0 0-2 93-92-156-156-156 156 94 92 0 2-250 0 0-250 0 0 93 93 157-156-157-156-93 94 0 0 0-250 250 0 0 0-94 93 156 157 156-157-93-93 0 0 250 0 0 250z',
+          'transform': 'matrix(1 0 0 -1 0 850)'
+        },
+        click: () => {
+          this.resetGraphToDefault();
+          if (this.assetCode !== undefined) {
+            this.getAssetReadings(this.payload);
+          }
+        }
+      }
+    ]]
+  };
+
+  payload = {
+    assetCode: '',
+    start: 0,
+    len: 600,
```

Comment:
I would be better to put 600, i.e. Max Datapoints, in a variable and use that variable throughout the file.